music21/environment.py

```diff
@@ -1566,31 +1584,28 @@ def testEnvironmentA(self):
         self.assertEqual(list(env['localCorpusSettings']), ['/a', '/b'])
 
     @unittest.skipUnless(
-        os.access(Environment().getDefaultRootTempDir(), stat.S_IRWXU),
-        'test will programmatically set read/write/exec permissions on this dir'
-    )
-    @unittest.skipIf(
-        common.getPlatform() == 'win',
-        'os.chmod does not have the intended effect on Windows'
```

Comment:
Love that this test runs on GitHub actions now. But now it won't run on Macs. Can we add a different assertion for Macs inside this test so that we can still only skip this test if windows? Hoping it's just a matter of checking platform and then making a different assertion, not causing you to rewrite the permissions fiddling.

Reply:
This test fail on Mac because the permissions fiddling behaves differently between Unix and other platforms (Mac and Windows). Should we include Mac into the routine of permissions fiddling same as Unix? Or just keep the current one?

Reply:
I get this result if I run the test as written:
```
AssertionError: 'music21-userid-501' not found in '/var/folders/sd/zw9wtxl552j6wn4yvt1nhg5r0000gn/T/music21-2_cvthvx'
```
Can't we just assert something else if system.platform() == 'darwin'? Or is your failure different?

Reply:
Also, `os.getuid()` is available on Mac, because when Python docs say UNIX they include Mac.

```
>>> import os
>>> os.getuid()
501
```

Reply:
>　Can't we just assert something else if system.platform() == 'darwin'?

Yeah. That's what I mean for "tinny modification". OK. I will do it.

Reply:
I'm still confused -- I thought you were going to leave the behavior alone and just change the test to assert the getUid() version for Linux and the 8-hashed characters for Mac. But you left the assertions alone, widened the test to run on Macs also, and then changed the behavior to use getUid() for Linux and Mac. I think that's fine, it's just not what I expected you to do based on what you said. If we keep it this way -- again, fine with me -- then we should update the docs above, where you say only Linux platforms have the getUid() version and other platforms have 8 hashed characters. Thanks.

Reply:
> and if possible, refactor out anything specific to unix distribution to a separate function/method

I agree with Michael's impulse to factor any Linux workarounds out so that they are easier to maintain (or ignore). But @tanchihpin0517 identified some two methods that weren't cleanly separated, so upon refactoring there really isn't much that's Linux specific anymore. What's different is that Windows doesn't have `getuid()`. Mac users are less likely than Linux users to hit permission errors, but we should still try to catch the permission errors and put user-friendly names on the fallback dirs we create. Just thinking out loud, not recommending any changes.

Reply:
> I'm still confused -- I thought you were going to leave the behavior alone and just change the test to assert the getUid() version for Linux and the 8-hashed characters for Mac.

It's my fault. I accepted your recommendation, doing permission fiddling on Unix (including Mac) rather than only on Linux, but didn't change the doc. I think that's why you feel confused.